lib/karto_sdk/src/Mapper.cpp

```diff
@@ -1989,6 +2028,18 @@ void MapperGraph::CorrectPoses()
   }
 }
 
+void MapperGraph::UpdateLoopScanMatcher(kt_double rangeThreshold)
+{
+  if (m_pLoopScanMatcher) {
+    delete m_pLoopScanMatcher;
+  }
+  m_pLoopScanMatcher = ScanMatcher::Create(m_pMapper,
```

Comment:
How are these https://github.com/SteveMacenski/slam_toolbox/blob/aab910110252d732dbcfd6d70f7ad3cc2b570b90/lib/karto_sdk/include/karto_sdk/Mapper.h#L1492-L1497 states updated for the new one (or is it necessary?)

Reply:
Particularly
```
  std::pair<kt_double, Pose2> * m_pPoseResponse;
  std::vector<kt_double> m_xPoses;
  std::vector<kt_double> m_yPoses;
```

Those look to me from a first glance to be lost that may be necessary to keep

Reply:
Actually - after looking they appear to only be used locally in the scan matcher and deleted / cleared on new use - can you verify for me that that's true?

Reply:
I have thought about the same thing here and have also come to the conclusion that the data you mentioned is overwritten here in every call anyway.
In my tests with this fix I have not seen any issues so far that would indicate otherwise, but I must admit that this is something I am still not fully comfortable with.

What do you have in mind for verification?

Reply:
I think we both came to the same conclusion - these are OK to override